e2etest/newe2e_task_runazcopy_login_logout.go

```diff
@@ -0,0 +1,76 @@
+package e2etest
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+var _ AzCopyStdout = &AzCopyInteractiveStdout{}
+
+// AzCopyInteractiveStdout is still a semi-raw stdout struct.
+type AzCopyInteractiveStdout struct {
+	Messages []string
+	asserter Asserter
+}
+
+// NewInteractiveWriter creates a new InteractiveWriter instance.
+func NewAzCopyInteractiveStdout(a Asserter) *AzCopyInteractiveStdout {
+	return &AzCopyInteractiveStdout{
+		asserter: a,
+	}
+}
+
+func (a *AzCopyInteractiveStdout) RawStdout() []string {
+	return a.Messages
+}
+
+func (a *AzCopyInteractiveStdout) Write(p []byte) (n int, err error) {
+	str := string(p)
+	lines := strings.Split(strings.TrimSuffix(str, "\n"), "\n")
+	n = len(p)
+
+	for _, v := range lines {
+		a.Messages = append(a.Messages, v)
+		a.asserter.Log(v)
+	}
+
+	return
+}
+
+func (a *AzCopyInteractiveStdout) String() string {
+	return strings.Join(a.RawStdout(), "\n")
+}
+
+func RunAzCopyLoginLogout(a Asserter, verb AzCopyVerb) AzCopyStdout {
```

Comment:
This is also interesting to me-- Is there a reason to prefer a separate function over integrating with RunAzCopy?

Reply:
I did this mostly because login needs some interactive stdout, whereas RunAzCopy is just reading the stdout. There is probably a clean way to integrate it with the other method. this can be cleaned this up after the PR is merged